cnd/tests/ethereum_missing_blocks.rs

```diff
@@ -1,13 +1,14 @@
 pub mod ethereum_helper;
```

Comment:
I think I did actually fix the problem you are trying to tackle here, but for some reason the changes I made have been lost and are not part of my branch `resume-swap-rebased`.

I believe you are trying to avoid the problem of running out of blocks to look in the past, prompting an error. We don't actually care about old blocks in this test, since that is tested elsewhere, so what I did was to add to the hashmap of `all_blocks` a `block_0_with_arbitrary_older_timestamp.json` whose hash matches `block1.parent_hash` and ensure that the `start_of_the_swap` is set to be younger than the timestamp of `block_0_with_arbitrary_older_timestamp.json`. Does that make sense?

Reply:
Yes that makes sense, then the `start_of_time` in the code will need some explanation for how it links to the test data.  I'll have a go at both this and what I outlined above and see which one reads the cleanest.  Cheers man.

Reply:
I did the lazy thing; since I had already moved the transaction from block 2 to block 3 in all the test data I just left it there and included block 1 in the 'all blocks' vector.  This does have the effect, IIRC, that now the 'big gap' is one block less.